src/commons/asBaseComponent.tsx

```diff
@@ -21,31 +21,21 @@ type ThemeComponent = {
 
 function asBaseComponent<PROPS, STATICS = {}>(WrappedComponent: React.ComponentType<any>): React.ComponentClass<PROPS & ThemeComponent> & STATICS {
   class BaseComponent extends UIComponent {
-    state = Modifiers.generateModifiersStyle(undefined, BaseComponent.getThemeProps(this.props, this.context));
     static displayName: string | undefined;
     static propTypes: any;
     static defaultProps: any;
 
-    static getDerivedStateFromProps(nextProps: any, prevState: any) {
-      const themeProps = BaseComponent.getThemeProps(nextProps, undefined);
-      const newModifiers = Modifiers.generateModifiersStyle(undefined, themeProps);
-      if (!_.isEqual(newModifiers, prevState)) {
-        return newModifiers;
-      }
-
-      return null;
-    }
-
     static getThemeProps = (props: any, context: any) => {
       return Modifiers.getThemeProps.call(WrappedComponent, props, context);
     }
 
     render() {
       const themeProps = BaseComponent.getThemeProps(this.props, this.context);
+      const modifiers = Modifiers.generateModifiersStyle(undefined, themeProps);
```

Comment:
Is it worth it to memoize the result?

Reply:
I thought about it in the past. 
But I think memoizing it will be too much since there can so many variations of props.